code/game/objects/items/weapons/mop.dm

```diff
@@ -43,7 +43,7 @@ obj/item/weapon/mop/proc/clean(turf/simulated/A)
 	if(istype(turf))
 		user.visible_message("<span class='warning'>[user] begins to clean \the [turf] with [src].</span>")
 
-		if(do_after(user, mopspeed))
+		if(do_after(user, mopspeed, target = src))

```

Comment:
Do you even know what `src` here is? This will put the progress bar in nullspace, change it to `target = turf`